### PR TITLE
security: disable plugin in default and persist file in specified dir

### LINKED ISSUE #7094: security: disable plugin in default and persist file in specified dir
## Bug Report
There are some leaks to inject errors.
- load mistake plugin anywhere
- no need to expose the persist interface, we just use it in  DR_STATS

### What did you expect to see?
- Load plugin and persist file in specified dir [config data dir]
- Judge persist file name rather than delete this interface to keep compatible
- support a config to decide whether to enable the plugin

### What did you see instead?
some leaks

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #7094

### What is changed and how does it work?

```commit-message
```
- Load plugin and persist file in specified dir [config data dir]
- Judge persist file name rather than delete this interface to keep compatible
- Support a config to decide whether to enable the plugin

### Check List

Tests

- Unit test

### Release note

```release-note
Add compile configuration to enable `Load Plugin` and disable `Load Plugin` by default. Need to `make pd-server-plugin` when want to enable plugin
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pkg/schedule/config/config.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ const (
 	defaultEnablePlacementRules  = true
 	defaultEnableWitness         = false
 	defaultHaltScheduling        = false
+	defaultEnableSchedulePlugin  = false
```
Comment (lhy1024) [inline pkg/schedule/config/config.go:60]: Maybe we need to add a release note about the default value changed

Comment (HuSharp) [reply pkg/schedule/config/config.go:60]: added.

File: server/api/plugin.go [hunk 2]
```diff
@@ -48,6 +48,10 @@ func newPluginHandler(handler *server.Handler, rd *render.Render) *pluginHandler
 // @Failure  500  {string}  string  "PD server failed to proceed the request."
 // @Router   /plugin [post]
 func (h *pluginHandler) LoadPlugin(w http.ResponseWriter, r *http.Request) {
+	if !h.GetScheduleConfig().EnableSchedulePlugin {
```
Comment (nolouch) [inline server/api/plugin.go:51]: I think it's better to be a `compile option`. otherwise, hackers may enable it.

Comment (HuSharp) [reply server/api/plugin.go]: after `make pd-server-plugin`:
```bash
curl -X POST 'http://127.0.0.1:2379/pd/api/v1/plugin'                                                                                                                                                                
load plugin is disabled, please `make pd-server PLUGIN=1` first%

curl -X DELETE 'http://127.0.0.1:2379/pd/api/v1/plugin'                                                                                                                                                              
unload plugin is disabled, please `make pd-server PLUGIN=1` first%
```

**Comment by HuSharp** [comment]
friendly ping :) @nolouch @lhy1024

File: server/api/plugin.go (hunk 2 above)
Comment (lhy1024) [reply server/api/plugin.go]: how about `PLUGIN=1 make pd-server`, similiar to `SWAGGER` and `DASHBOARD`

File: Makefile [hunk 3]
```diff
@@ -82,6 +86,9 @@ pd-server-failpoint:
 	FAILPOINT=1 $(MAKE) pd-server || { $(FAILPOINT_DISABLE); exit 1; }
 	@$(FAILPOINT_DISABLE)
 
+pd-server-plugin:
+	PLUGIN=1 $(MAKE) pd-server
+
```
Comment (HuSharp) [inline Makefile:91]: Do you mean remove this command instead of original `PLUGIN=1 $(MAKE) pd-server`? @lhy1024

Comment (lhy1024) [reply Makefile:91]: yes

**Review by lhy1024 — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**